File: GridView/app/src/main/java/com/techmania/gridview/MainActivity.java
package com.techmania.gridview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.GridView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    GridView girdView;
    ArrayList<String> text = new ArrayList<>();
    ArrayList<Integer> image = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        girdView = findViewById(R.id.gridView);
        fillArray();
        GridAdapter adapter = new GridAdapter(this,text,image);
        girdView.setAdapter(adapter);

        girdView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(getApplicationContext(),"You selected "+text.get(position),Toast.LENGTH_SHORT).show();
            }
        });
    }

    public void fillArray(){
        text.add("Butterfly");
        text.add("Cat");
        text.add("Flower");
        text.add("Hippo");
        text.add("Monkey");
        text.add("Mushroom");
        text.add("Panda");
        text.add("Rabbit");
        text.add("Raccoon");

        image.add(R.drawable.butterfly);
        image.add(R.drawable.cat);
        image.add(R.drawable.flower);
        image.add(R.drawable.hippo);
        image.add(R.drawable.monkey);
        image.add(R.drawable.mushroom);
        image.add(R.drawable.panda);
        image.add(R.drawable.rabbit);
        image.add(R.drawable.raccoon);
    }
}

File: GridView/app/src/main/java/com/example/gridview/MainActivity.kt
package com.example.gridview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.GridView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    lateinit var gridView : GridView
    var nameList = ArrayList<String>()
    var imageList = ArrayList<Int>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        gridView = findViewById(R.id.gridView)
        fillArrays()
        val adapter = AnimalAdapter(this,nameList,imageList)

        gridView.adapter = adapter

        gridView.setOnItemClickListener { adapterView, view, position, id ->
            Toast.makeText(applicationContext,"You selected the ${nameList[position]}",Toast.LENGTH_SHORT).show()
        }
    }

    fun fillArrays() {
        nameList.add("Butterfly")
        nameList.add("Cat")
        nameList.add("Flower")
        nameList.add("Hippo")
        nameList.add("Monkey")
        nameList.add("Mushroom")
        nameList.add("Panda")
        nameList.add("Rabbit")
        nameList.add("Raccoon")

        imageList.add(R.drawable.butterfly)
        imageList.add(R.drawable.cat)
        imageList.add(R.drawable.flower)
        imageList.add(R.drawable.hippo)
        imageList.add(R.drawable.monkey)
        imageList.add(R.drawable.mushroom)
        imageList.add(R.drawable.panda)
        imageList.add(R.drawable.rabbit)
        imageList.add(R.drawable.raccoon)
    }
}